src/material/datepicker/year-view.ts

```diff
@@ -198,6 +198,25 @@ export class MatYearView<D> implements AfterContentInit, OnDestroy {
     );
   }
 
+  /** Handles when a new month becomes active. */
```

Comment:
This is nearly identical to the `_monthSelected` method above. I'm open to suggestions on how to refactor this, or if we should refactor it to share code.